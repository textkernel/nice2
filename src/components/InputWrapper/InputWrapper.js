import React from 'react';
import PropTypes from 'prop-types';
import bem from 'bem';
import { Button } from '../../index';
import styles from './InputWrapper.scss';

const { block, elem } = bem({
    name: 'InputWrapper',
    classnames: styles,
    propsToMod: ['isFocused']
});

const InputWrapper = React.forwardRef((props, ref) => {
    const handleClear = e => {
        const { onClear } = props;

        e.stopPropagation();
        if (onClear) {
            onClear(e);
        }
    };

    const { children, clearLabel, showClearButton, onClear, refKey, isFocused, ...rest } = props;

    return (
        <div ref={ref} {...rest} {...block(props)}>
            {children}
            {showClearButton && !isFocused && (
                <Button
                    isInline
                    context="link"
                    onClick={handleClear}
                    {...elem('clearButton', props)}
                >
                    {clearLabel}
                    <span>&times;</span>
                </Button>
            )}
        </div>
    );
});

InputWrapper.displayName = 'InputWrapper';

InputWrapper.propTypes = {
    /** show Clear button on hover even if there are no selectedSuggestions passed */
    showClearButton: PropTypes.bool,
    /** clear button label */
    clearLabel: PropTypes.string,
    /** reset the selected suggestions array to it's default value */
    onClear: PropTypes.func,
    /** is content should be considered to be in focus, when clear button should not be shown */
    isFocused: PropTypes.bool
};

InputWrapper.defaultProps = {
    showClearButton: false,
    clearLabel: '',
    onClear: null,
    isFocused: false
};

export default InputWrapper;
