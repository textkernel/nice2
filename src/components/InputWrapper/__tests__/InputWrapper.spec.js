import React from 'react';
import toJson from 'enzyme-to-json';
import InputWrapper from '../InputWrapper';

describe('InputWrapper', () => {
    it('should render correctly', () => {
        const wrapper = mount(<InputWrapper>some children</InputWrapper>);

        expect(toJson(wrapper)).toMatchSnapshot();
        expect(wrapper.find('button')).toHaveLength(0);
    });
    it('should add clear button if showClearButton is true', () => {
        const wrapper = mount(
            <InputWrapper showClearButton clearLabel="Clear">
                some children
            </InputWrapper>
        );

        expect(toJson(wrapper)).toMatchSnapshot();
        expect(wrapper.find('button')).toHaveLength(1);
    });
    it('should call onClear callback correctly', () => {
        const onClearMock = jest.fn();
        const wrapper = mount(
            <InputWrapper showClearButton clearLabel="Clear" onClear={onClearMock}>
                tag
            </InputWrapper>
        );

        wrapper.find('button').simulate('click');

        expect(onClearMock).toHaveBeenCalledTimes(1);
    });
});
