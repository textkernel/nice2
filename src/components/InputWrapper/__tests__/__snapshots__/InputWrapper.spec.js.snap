// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputWrapper should add clear button if showClearButton is true 1`] = `
<InputWrapper
  clearLabel="Clear"
  isFocused={false}
  onClear={null}
  showClearButton={true}
>
  <div
    className="InputWrapper"
  >
    some children
    <Button
      className="InputWrapper__clearButton"
      context="link"
      disabled={false}
      href={null}
      isBlock={false}
      isInline={true}
      onClick={[Function]}
      size="normal"
      type="button"
    >
      <button
        className="Button Button--context_link Button--isInline InputWrapper__clearButton"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Clear
        <span>
          Ã—
        </span>
      </button>
    </Button>
  </div>
</InputWrapper>
`;

exports[`InputWrapper should render correctly 1`] = `
<InputWrapper
  clearLabel=""
  isFocused={false}
  onClear={null}
  showClearButton={false}
>
  <div
    className="InputWrapper"
  >
    some children
  </div>
</InputWrapper>
`;
